### PR TITLE
CMake addition

### PR DESCRIPTION
**Pull request by larivierec**
The GemParticles can now make use of the classes in AntTweakBar.
Using this library we will be able to modify OpenGL properties during runtime.

The application will compile AntTweakBar and link it against the application.
If sources are needed, they may be included within the GemParticles application.

### CODE REVIEW

**Comment by frtru** [comment]
Can't merge, gotta fix conflicts first. But, how is GemParticlesGUI going to use entities from GemParticles? Is GemParticlesGUI going to define stricly helper classes/functions for us to call in GemParticles?

**Review by frtru — COMMENTED** [review]
Everything looks good, but I'd to keep the VS files just as backup for now.

**Comment by larivierec** [comment]
i'll have to update the cmake files first. until then this shouldn't be merged.

**Comment by larivierec** [comment]
https://github.com/lighttransport/nanogi/blob/master/cmake/FindGLFW.cmake 

https://github.com/nvpro-samples/shared_sources/blob/master/cmake/FindGLFW.cmake

Use these as a reference 
for CPU architecture

File: src/particle_system.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*************************************************************************
```
Comment (frtru) [inline src/particle_system.cc]: You can remove this file, it is not needed anymore. Everything is in .hh, because of templates.

**Review by frtru — CHANGES_REQUESTED** [review]
I'd rather keep the VS files and I'll update them so that they can coexist and not affect the code. The reason is that, I don't think it works the way I would expect it to work on windows and it would be too much troubles to add a file to the cmakelists.txt and regenerate the visual studio files each time, since right now I have to manually edit a bunch of variables that give me errors even though I have the XXX_HOME variables defined. Even with that I have some weird compilation/linking errors. 

So just re-add the VS files, delete particle_system.cc and we're good to go :)

**Pull request merged**